### PR TITLE
pull request #3 from tenyiting/sign_up

### PR DESCRIPTION
**Pull request by tenyiting**
ready

### CODE REVIEW

**Comment by tenyiting** [comment]
ready

File: app/assets/stylesheets/users.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (huongnt-2545) [inline app/assets/stylesheets/users.scss:1]: delete this file if it not useful

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module UsersHelper
+  def gravatar_for user
+   gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
```
Comment (huongnt-2545) [inline app/helpers/users_helper.rb:3]: wrong indent
```suggestion
    gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}"
    image_tag(gravatar_url, alt: user.name, class: "gravatar")
  end
```

File: app/views/shared/_error_messages.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<% if @user.errors.any? %>
+<div id="error_explanation">
```
Comment (huongnt-2545) [inline app/views/shared/_error_messages.html.erb:2]: wrong indent in this file, please check

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @user, url: signup_path do |f| %>
+      <% if @user.errors.any? %>
+        <div id="error_explanation">
+          <div class="alert alert-danger">
+            <%= t (:the_form_contains)%>
+            <%= pluralize @user.errors.count, t(:error) %>.
+          </div>
+          <ul>
+          <% @user.errors.full_messages.each do |msg| %>
```
Comment (huongnt-2545) [inline app/views/users/new.html.erb:13]: wrong indent
```suggestion
            <% @user.errors.full_messages.each do |msg| %>
               <li><%= msg %></li>
            <% end %>
```

File: config/routes.rb [hunk 5]
```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
+  get 'users/show'
```
Comment (huongnt-2545) [inline config/routes.rb:2]: remove line 2, 3. It's ready defined at line 10 `resources :users, only: %i(new create show)`

File: test/controllers/users_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/users_controller_test.rb:1]: delete this file

**Review by huongnt-2545 — COMMENTED** [review]
Please create new commit for each branch, you are using previous merged commit from other brach

**Comment by tenyiting** [comment]
Ready

File: app/views/users/new.html.erb (hunk 4 above)
Comment (ngocvt-0484) [reply app/views/users/new.html.erb:13]: not fix yet

**Comment by tenyiting** [comment]
Ready

**Pull request merged**